presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -363,6 +373,40 @@ void PrestoServer::run() {
         json infoStateJson = convertNodeState(server->nodeState());
         http::sendOkResponse(downstream, infoStateJson);
       });
+  httpServer_->registerPut(
+    "/v1/info/state",
+    [server = this](
+        proxygen::HTTPMessage* /*message*/,
+        const std::vector<std::unique_ptr<folly::IOBuf>>& body ,
+        proxygen::ResponseHandler* downstream) {
+         std::string requestBody=getRequestBody(body);
+         try {
+            folly::dynamic jsonBody = folly::parseJson(requestBody);
+            if (jsonBody == "SHUTTING_DOWN") {
+                // Print message and initiate shutdown
+                LOG(INFO) << "Shutdown requested";
+                if (server->nodeState() == NodeState::kActive) {
+                    // Call stop() to handle the shutdown process
+                    std::thread([server]() {
+                            server->stop();
+                        }).detach();
+                } else {
+                    // Print message and indicate that shutdown is already requested
+                    LOG(INFO) << "Shutdown already requested";
+                }
+                http::sendOkResponse(downstream);
+            } else {
+                // Throw a 400 Bad Request error if the body has some other value
+                http::sendErrorResponse(
+                    downstream,
+                    "Bad Request: Body must contain 'SHUTTING_DOWN'.",http::kHttpBadRequest);
```

Comment:
Can you check the Java behavior for this case?

Reply:
I get the following response for presto java:
<img width="1728" alt="Screenshot 2024-10-04 at 11 00 59 AM" src="https://github.com/user-attachments/assets/0b0fca85-eeee-498b-8a99-1e9582dc3c8a">

It seems that we directly get the error stack. Should I remove "Bad Request:..." message ?

Reply:
No, this implementation is better.